### PR TITLE
pdf-sign: Fix program name in help text

### PR DESCRIPTION
**Pull request by svenssonaxel**
## Description of changes
Use correct program name, rather than wrapped program, in usage text.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by svenssonaxel** [comment]
@TomaSajt Please review.

**Review by TomaSajt — COMMENTED** [review]
Thanks for creating this PR, I didn't catch that the help menu was wrong.

Your solution uses `sed`, however I think that using `substituteInPlace` with `--replace-fail` is a bit better, since it will fail if there's nothing to substitute.

```
substituteInPlace $exe \
  --replace-fail "ArgumentParser(" "ArgumentParser(prog='$exe',"
```

something like this should work.

---

Another solution would be to keep the unwrapped scripts in a different directory, so that they can have the same name, thus not messing up the displayed name.

---

Also, please add a comment into the script saying why we're doing this replacement.

**Comment by svenssonaxel** [comment]
@TomaSajt Fantastic feedback. I opted for keeping the names as in your second suggestion. I considered it previously, but wasn't sure if this would break some kind of recommended practice.

File: pkgs/by-name/pd/pdf-sign/package.nix [hunk 1]
```diff
@@ -38,8 +38,11 @@ stdenv.mkDerivation {
     runHook preInstall
 
     for exe in "pdf-sign" "pdf-create-empty" "pdf-from-text"; do
-      install -Dm755 $exe -t $out/bin
-      wrapProgram $out/bin/$exe --prefix PATH : ${binPath}
+      # Install wrapped programs into $out/lib so that they are not renamed.
+      # Renaming them, like wrapProgram does, would produce the wrong output
+      # from `--help`.
+      install -Dm755 $exe -t $out/lib
+      makeBinaryWrapper $out/lib/$exe $out/bin/$exe --prefix PATH : ${binPath}
```
Comment (drupol) [inline pkgs/by-name/pd/pdf-sign/package.nix:45]:
```suggestion
      makeWrapper $out/lib/$exe $out/bin/$exe --prefix PATH : ${binPath}
```

**Pull request merged**